File: API_Escola_Stable/Models/Matricula.cs
using System.ComponentModel.DataAnnotations;
//GABRIEL
namespace API_Uni_Func.Models
{
    public class Matricula
    {
        [Key]
        public int IdMatricula { get; set; }
        public int IdEstudante { get; set; }
        public Estudante? EstudanteMatriculado {  get; set; }
        public int IdCurso { get; set; }
        public Curso? CursoMatriculado { get; set; }
        public DateTime? DataMatricula { get; set; }
        public bool MatriculaAtiva {  get; set; }
        public int CronogramaId { get; set; }
        public Cronograma? CronogramasEstudante { get; set; }
    }
}

File: AppUniversidade/src/app/Matricula.ts
import { Cronograma } from "./Cronograma";
import { Curso } from "./Curso";
import { Estudante } from "./Estudante";

export class Matricula{
    IdMatricula: number = 0;
    IdEstudante: number = 0;
    EstudanteMatriculado: Estudante | null = null;
    IdCurso: number = 0;
    CursoMatriculado: Curso | null = null;
    DataMatricula: Date | undefined;
    MatriculaAtiva: boolean | undefined;
    CronogramaId: number = 0;
    CronogramaEstudante: Cronograma | null = null;

}